hot-reload-core/src/main/kotlin/org/jetbrains/compose/reload/core/Task.kt

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2024-2025 JetBrains s.r.o. and Compose Hot Reload contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+ */
+
+package org.jetbrains.compose.reload.core
+
+import java.util.concurrent.RejectedExecutionException
+import java.util.concurrent.atomic.AtomicReference
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+import kotlin.coroutines.createCoroutine
+import kotlin.coroutines.resume
+
+private val logger = createLogger()
+
+public interface Task<T> : Future<T>, CoroutineContext.Element {
+    public val name: String?
+    public val value: Future<T>
+    public val onStop: Future<Nothing>
+    public fun stop(exception: Throwable? = null): Boolean
+
+    public fun <T> subtask(
+        name: String? = null, context: CoroutineContext = EmptyCoroutineContext, body: suspend Task<T>.() -> T
+    ): Task<T>
+
+    override val key: CoroutineContext.Key<*> get() = Key
+
+    public companion object Key : CoroutineContext.Key<Task<*>>
+}
+
+
+public fun Task<*>.isActive(): Boolean = !value.isCompleted() && !isStopped()
```

Comment:
We discussed during the demo that maybe this should be suspendable. You decided against it?